backend/app/repositories/memory.py

```
@@ -300,6 +341,61 @@ def cancel(self, booking_id: str, farmer_id: str) -> Optional[Dict[str, Any]]:
+                != first_slot
+                for data in data_list[1:]
+            ):
+                return None
+            slot_key = _slot_key(first_data["centre_id"], first_data["slot_date"], first_data["slot_window"])
+            current = self._active_counts.get(slot_key, 0)
+            
+            if current + len(booking_ids) > capacity:
+                return None
+
+            created_records = []
+            for bid, data in zip(booking_ids, data_list):
```

Ruff (0.16.3)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)